contracts/identity/lib.rs

```diff
@@ -186,13 +194,40 @@ mod identity {
 				identity_count: 0,
 				network_name: Default::default(),
 				network_id_counter: 0,
+				recovery_account_of: Default::default(),
 				admin: caller,
 			}
 		}
 
+		/// Returns the `IdentityInfo` of an identity that is associated with
+		/// the provided `IdentityNo`.
+		#[ink(message)]
+		pub fn identity(&self, identity_no: IdentityNo) -> Option<IdentityInfo> {
+			self.number_to_identity.get(identity_no)
+		}
+
+		/// Returns the owner of an identity.
+		#[ink(message)]
+		pub fn owner_of(&self, identity_no: IdentityNo) -> Option<AccountId> {
+			self.owner_of.get(identity_no)
+		}
+
+		/// Returns the owner of an identity.
+		#[ink(message)]
+		pub fn identity_of(&self, owner: AccountId) -> Option<IdentityNo> {
+			self.identity_of.get(owner)
+		}
+
+		/// Returns the number of identities that exist.
+		#[ink(message)]
+		pub fn identity_count(&self) -> IdentityNo {
```

Comment:
This is not correct.

Reply:
Are you referring to the return type since the `identity_count` is a `u32`, or?

Reply:
`self.identity_count` doesn't actually store the number of identities. It's becausewe didn't decrease `self.identity_count` in `remove_identity`.

it is used for assigning unique `identity_no`s to newly created identities and it's the same as number of identities created so far.

Reply:
Yes, that is correct. There can be two solutions to this:
1. I add a NOTE to the function comment so that it explains this
2. We rename the `identity_count` to `next_identity_no`

In any case, I think we should also change the type of this storage value to an `IdentityNo`

Reply:
agreed
and we need a storage item for the number of existing identities

Reply:
Would having a storage item for the number of existing identities be useful? I am not sure we need that.

Reply:
i don't think we really need `identity_count`

Reply:
Are you suggesting I rename it to something like `latest_identity_no`?